src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -109,6 +110,55 @@
 (def OIDMappings
   {schema/Str schema/Keyword})
 
+(def default-cadir
+  "/etc/puppetlabs/puppetserver/ca")
+
+(defn default-cacert
+  [cadir]
+  (str cadir "/ca_crt.pem"))
+
+(defn default-cacrl
+  [cadir]
+  (str cadir "/ca_crl.pem"))
+
+(defn default-cakey
+  [cadir]
+  (str cadir "/ca_key.pem"))
+
+(defn default-capub
+  [cadir]
+  (str cadir "/ca_pub.pem"))
+
+(defn default-cert-inventory
+  [cadir]
+  (str cadir "/inventory.txt"))
+
+(defn default-csrdir
+  [cadir]
+  (str cadir "/requests"))
+
+(defn default-signeddir
+  [cadir]
+  (str cadir "/signed"))
+
+(defn default-serial
+  [cadir]
+  (str cadir "/serial"))
+
+(schema/defn ^:always-validate
+initialize-ca-config
+  [ca-data-structure]
+  (-> ca-data-structure
+      (update-in [:cadir] #(or % default-cadir))
```

Comment:
You can, and should if you keep this structure, just use `update` here and remove the `[]` from the key.

You _might_ also want to try setting the `cadir` to the desired value and then creating a map of defaults and merging them?

You can also use argument destructuring to get the value from a map or a default [see](https://clojure.org/guides/destructuring#_keyword_arguments).

This is something I would write:
```clj
(defn init-ca
  [{:keys [cadir] :or {cadir default-cadir} :as ca-data}]
  (let [defaults {:cadir cadir
                  :cacert (default-cacert cadir)
                  :cacrl (default-cacrl cadir)
                  ;; ...
                 }]
    (merge defaults ca-data)))
```

But I think the main thing here is readability, I think lifting the values to a map at the top of the function brings them into focus and removes as much ceremony around setting them as possible. I'd say experiment with the approaches until you find something you think someone w/o a lot of clojure experience will easily understand in a year or two's time (I could see, based on that, making an argument against destructuring).